lexicon/providers/mythicbeasts.py

```diff
@@ -0,0 +1,280 @@
+
+    # Create record. If record already exists with the same content, do nothing'
+    def _create_record(self, rtype, name, content):
+        LOGGER.debug("type %s", rtype)
+        LOGGER.debug("name %s", name)
+        LOGGER.debug("content %s", content)
+
+        if rtype == 'CNAME':
+            content = self._fqdn_name(content)
+
+        data = {
+            "records": [
+                {
+                    "host": self._relative_name(name),
+                    "type": rtype,
+                    "data": content,
+                }
+            ]
+        }
+        if self._get_lexicon_option("ttl"):
+            data["records"][0]["ttl"] = self._get_lexicon_option("ttl")
+
+        payload = {"success": True}
+        try:
+            payload = self._post(f"/zones/{self.domain}/records", data)
+        except requests.exceptions.HTTPError as err:
+            if err.response.status_code != 400 and err.response.json()["errors"][0][0:16] != 'Duplicate record':
+                raise
+
+        if rtype == 'A' or rtype == "TXT":
+            # need to wait and poll here until verified that DNS change is live
+            changes_are_live = False
+            timeout = 300
+            start = time.time()
+
+            while(not changes_are_live and time.time() - start < timeout):
+                try:
+                    self._get(f"/zones/{self.domain}/records/{self._relative_name(name)}/{rtype}?verify")
+                    changes_are_live = True
+                except requests.exceptions.HTTPError as err:
+                    if err.response.status_code != 409:
+                        raise
+                    else:
+                        time.sleep(10)
+
+            if(not changes_are_live):
+                raise Exception("Timed out trying to verify changes were live")
+
+        if "message" in payload:
+            return payload["message"]
+        elif "success" in payload:
+            return payload["success"]
+
+    # List all records. Return an empty list if no records found
+    # type, name and content are used to filter records.
+    # If possible filter during the query, otherwise filter after response is received.
+    def _list_records(self, rtype=None, name=None, content=None):
+        filter_obj = {}
+        if rtype:
+            filter_obj["type"] = rtype
+        if name:
+            filter_obj["host"] = self._relative_name(name)
+        if content:
+            filter_obj["data"] = content
+
+        records = []
+        payload = self._get(f"/zones/{self.domain}/records", filter_obj)
+
+        LOGGER.debug("payload: %s", payload)
+
+        for record in payload["records"]:
+            processed_record = {
+                "type": record["type"],
+                "name": self._full_name(record["host"]),
+                "ttl": record["ttl"],
+                "content": record["data"],
+                # no id is available, so we need to make our own
+                "id": hashlib.md5(
```

Comment:
Here also I can propose you a code to generate a "short" indentifier that is used in several modules where ids are not provided by the API: https://github.com/AnalogJ/lexicon/blob/59419ffba7797a860251c83f73221054b3f72862/lexicon/providers/azure.py#L361-L365